Adds PR template

ale8k opened the pull request.

mhilton commented:
Unfortunately the equivalent for usernames is a significantly more involved change, so it is unlikely that will be able to change in the short term.

Commit:
Adds PR template

Commit:
Update templates

- Updates PR template to comment out optional sections.
- Adds a few standardised issue templates, namely bug, feature and support.

Commit:
Remove excess returns

alesstimec commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,23 @@
+- [ ] Covered by integration tests
+- [ ] Independent change*
+
+ _* For example, it does not require an sfproxy release before it can be merged._
```
Please remove this comment: candid has nothing to do with sfproxy

alesstimec commented on .github/ISSUE_TEMPLATE/support.md:
```diff
@@ -0,0 +1,6 @@
+# Issue: Support
+
+## Ask your question clearly and concisely (Opportunity for detail is below)
```
hmm.. not sure about the wording here.. seems like we don't trust our customers to provide a concise issue title..

ale8k replied:
That's fair, will leave it at 'Ask your question', I taken inspiration from a couple existing templates but I think you're right.

alesstimec commented on .github/ISSUE_TEMPLATE/feature.md:
```diff
@@ -0,0 +1,11 @@
+# Issue: Feature
+> Note: The quickest way to see a new feature would be to submit a pull request, which will be reviewed when possible.
+## What feature?
```
s/What feature/Feature description/

alesstimec commented on .github/ISSUE_TEMPLATE/feature.md:
```diff
@@ -0,0 +1,11 @@
+Please describe what functionality this feature pertains to.
+-->
+
+## Why include this feature?
```
I don't think we need this section. I believe users will describe the "why" in the feature description. We shouldn't ask them why they "feel" this feature should be included. Instead we should be interested in clear benefits of this feature.

alesstimec commented on .github/ISSUE_TEMPLATE/feature.md:
```diff
@@ -0,0 +1,11 @@
+# Issue: Feature
+> Note: The quickest way to see a new feature would be to submit a pull request, which will be reviewed when possible.
```
strictly speaking this is not absolutely true :) we have an architecture of candid in mind and outside users might not understand this fully. So a PR might lead to lengthy discussions an change requests. But i agree, we should not discourage anybody from proposing a PR. 

Instead, perhaps we should say something along the lines: "if you believe this feature requires a small change, feel free to propose a PR, which will be reviewed as soon as possible"

ale8k replied:
Ah ok, I'll update to your recommendation.

Commit:
Updates from comments

Commit:
Ales recommendation

Commit:
Additional comment for guidance.
